packages/test/test-version-utils/README.md

```diff
@@ -1,26 +1,137 @@
+a specific driver and different version combinations of Fluid API between layers via `TestObjectProvider` provided
+to the test. The different layers are loader, driver, container runtime and data runtime (includes DDS).  Version
+combinations and driver selection can be controlled via the command line for test set up using the provided
+ `describe*` functions.  For advanced usage, test suite can bypass the setup and command line and directly call our
+exports to get the versioned Fluid APIs.
+
+## Versioned combination test generation
+
+### Layer version combinations
+
+Similar to `mocha`'s `describe`, this package provide various `describe*` functions that will generate variants with
+a specific driver and different version combinations of Fluid API between layers. All possible layer combinations that
+are generated (empty entries are current versions):
+
+| Compat Kind         | Loader | Driver | Container Runtime | Data Runtime |
+| ------------------- | ------ | ------ | ----------------- | ------------ |
+| None                |        |        |                   |              |
+| Loader              |  old   |        |                   |              |
+| Driver              |        |  old   |                   |              |
+| ContainerRuntime    |        |        | old               |              |
+| DataRuntime         |        |        |                   | old          |
+| NewLoader           |        |  old   | old               | old          |
+| NewDriver           |  old   |        | old               | old          |
+| NewContainerRuntime |  old   |  old   |                   | old          |
+| NewDataRuntime      |  old   |  old   | old               |              |
+
+### Mocha test setup with layer version combinations
+
+There are three compat `describe*` to generate different combinations, depending of the need of the tests
+
+`describeFullCompat`: generate test variants with compat combinations that varies the version for all layers.
+
+- Used for tests that exercise all layers and will benefits compat combinations of all layers.
+
+`describeLoaderCompat`: generate test variants with compat combinations that only varies the loader version.
+
+- Use for tests that targets the loader layer, and don't care about compat combinations of other layers.
+- Test combination generated: [CompatKind.None, CompatKind.Loader]
+
+`describeNoCompat` - generate one test variant that doesn't varies version of any layers.
+
+- Use for tests that doesn't benefit or require any compat testing.
+- Test combination generated: [CompatKind.None]
+
+These compat `describe*` functions will also load the APIs with appropriate version and provide the test with a
+`TestObjectProvider` object, where the test can use to access Fluid functionality.  Even compat testing not
+necessary, `TestObjectProvider` provide convenience function and tracking capability, and it allows the test
```

Comment:
```suggestion
`TestObjectProvider` object, where the test can use to access Fluid functionality.  Even when compat testing
is not necessary, `TestObjectProvider` provides convenience function and tracking capability, and it allows the test
```

Can you reword "convenience function and tracking capability"? Not sure what it's about.

Reply:
will think about the reword.